apps/aecore/lib/aecore/peers/peer_connection.ex

```diff
@@ -0,0 +1,528 @@
+            peer = %{host: host, pubkey: r_pubkey, port: port, connection: self()}
+            :ok = do_ping(new_state)
+            Peers.add_peer(peer)
+            {:noreply, new_state}
+
+          {:error, _reason} ->
+            :gen_tcp.close(socket)
+            {:stop, :normal, state}
+        end
+
+      {:error, _reason} ->
+        {:stop, :normal, state}
+    end
+  end
+
+  def handle_info({:noise, _, <<@msg_fragment::16, n::16, m::16, fragment::binary()>>}, state) do
+    handle_fragment(state, n, m, fragment)
+  end
+
+  def handle_info({:noise, _, <<type::16, payload::binary()>>}, state) do
+    deserialized_payload = :erlang.binary_to_term(payload)
+    self = self()
+
+    case type do
+      @p2p_response ->
+        spawn(fn ->
+          handle_response(deserialized_payload, self, Map.get(state, :requests, :empty))
+        end)
+
+      @ping ->
+        spawn(fn -> handle_ping(deserialized_payload, self, state) end)
+
+      @get_header_by_hash ->
+        spawn(fn -> handle_get_header_by_hash(deserialized_payload, self) end)
+
+      @get_header_by_height ->
+        spawn(fn -> handle_get_header_by_height(deserialized_payload, self) end)
+
+      @get_n_successors ->
+        spawn(fn -> handle_get_n_successors(deserialized_payload, self) end)
+
+      @get_block ->
+        spawn(fn -> handle_get_block(deserialized_payload, self) end)
+
+      @get_mempool ->
+        spawn(fn -> handle_get_mempool(self) end)
+
+      @block ->
+        spawn(fn -> handle_new_block(deserialized_payload) end)
+
+      @tx ->
+        spawn(fn -> handle_new_tx(deserialized_payload) end)
+    end
+
+    {:noreply, state}
+  end
+
+  def handle_info({:tcp_closed, _}, state) do
+    Logger.info("Connection interrupted by peer - #{inspect(state)}")
+    Peers.remove_peer(state.r_pubkey)
+    {:stop, :normal, state}
+  end
+
+  defp do_ping(%{status: {:connected, socket}}) do
+    ping_object = local_ping_object()
+    serialized_ping_object = :erlang.term_to_binary(ping_object)
+    msg = <<@ping::16, serialized_ping_object::binary()>>
+    :enoise.send(socket, msg)
+  end
+
+  defp send_response(result, type, pid) do
+    payload =
+      case result do
+        {:ok, object} ->
+          %{result: true, type: type, reason: nil, object: object}
+
+        {:error, reason} ->
+          %{result: false, type: type, reason: reason, object: nil}
+      end
+
+    @p2p_response
+    |> pack_msg(payload)
+    |> send_msg_no_response(pid)
+  end
+
+  defp send_request_msg(msg, pid), do: GenServer.call(pid, {:send_request_msg, msg})
+
+  defp send_msg_no_response(msg, pid) when byte_size(msg) > @max_packet_size - 2 do
+    number_of_chunks = div(@fragment_size + byte_size(msg) - 1, @fragment_size)
+    send_chunks(pid, 1, number_of_chunks, msg)
+  end
+
+  defp send_msg_no_response(msg, pid), do: GenServer.call(pid, {:send_msg_no_response, msg})
+
+  defp send_chunks(pid, n, m, msg) when n == m do
```

Comment:
What does `n` and `m` stand for? Could they be more descriptive or add a comment?

Reply:
Peer messages are documented here https://github.com/aeternity/protocol/blob/30652d59858afb0663dfea95f2023a3254da0a6f/sync/p2p_messages.md - n is the current fragment index and m is the total fragments

Reply:
can you name it more descriptive anyways